src/runtime_cache_table.js

```diff
@@ -0,0 +1,34 @@
+#if CACHE_WASM_TABLE
+// Wrap the wasm table with a cache, to avoid repeatedly calling table.get().
+// That operation is fairly heavyweight in VMs.
+// N.B.: This cache assumes that the wasm does not modify the table, which is
+//       not necessarily true in relocatable builds (a side module segment may
+//       add items), or if we some day use wasm table operations in wasm.
+var originalWasmTable = wasmTable;
+wasmTable = {
+  _cache: [],
+  'length': wasmTable.length,
```

Comment:
`wasmTable.length` looks wrong here? Should this be `this._cache.length` instead?

(or alternatively if I was writing this golfing for code size, I'd write in `wasmTable._cache.length` to make Closure win three bytes replacing `this` with a likely single-letter variable, but that feels a bit conceptually dirty for the object to access itself via outside scope name, and could be confusing given the monkeypatch replacement above)

Reply:
actually, looks like this should be `'length': originalWasmTable.length`?

Reply:
Or maybe best would be `get 'length'() { return originalWasmTable.length }`, then the implementation of `grow` could simply be:

```js
'grow': function(by) {
     return originalWasmTable.grow(by);
}
```